subsys/bluetooth/host/classic/avdtp.c

```diff
@@ -425,7 +425,7 @@ static void avdtp_process_configuration_cmd(struct bt_avdtp *session, struct net
 					    uint8_t tid, bool reconfig)
 {
 	int err = 0;
-	int ret;
+	int l2cap_ret;
 	struct bt_avdtp_sep *sep;
 	struct net_buf *rsp_buf;
 	uint8_t error_code = 0;
```

Comment:
It's actually worse, since you have *three* error variables here with `error_code` included. I think you should be able to remove `l2cap_ret` completely and just reuse `err` when you need to store the value from `l2cap_chan_send()`. I'd also rename `error_code` to `avdtp_err`. At the end of the function when you need to decide about the state change, you shouldn't need to look at `err` at all, rather `!reconfig && avdtp_err == 0` is all that you need.

Reply:
Remove `l2cap_ret`. Change `error_code` as `avdtp_err_code`, keep `code` in the variable because the avdtp spec say it as `ERROR_CODE`.